src/recipe/recipe.service.ts

```diff
@@ -1,20 +1,18 @@
 import { Injectable, NotFoundException } from '@nestjs/common';
-import { PrismaService } from '../prisma/prisma.service';
 import { Recipe, Prisma } from '@prisma/client';
 import { UpdateRecipeRequest } from './dto';
 import { RecipeCacheService } from './recipe.cache.service';
+import { RecipeRepository } from './recipe.repository';
 
 @Injectable()
 export class RecipeService {
   constructor(
-    private prisma: PrismaService,
+    private readonly recipeRepository: RecipeRepository,
     private readonly recipeCacheService: RecipeCacheService,
   ) {}
 
   async createRecipe(data: Prisma.RecipeCreateInput): Promise<Recipe> {
```

Comment:
as per our discussion - perhaps you can get replace `Prisma.RecipeCreateInput` type with some DTO type?